### ISSUE TITLE
Experimental: Add a function to verify a signature for an address

### ISSUE THREAD

**Issue by mcintyre94** [solana-labs/solana-web3.js#2978]
## Motivation

In the `deserialize-transanction` example, we demonstrate how to verify the signature for a transaction

```ts
const signedByAddress = 'ED1WqT2hWJLSZtj4TtTdoovmpMrr7zpkUdbfxmcJR1Fq' as Address;
const signedBySignature = decodedTransaction.signatures[signedByAddress]!;
// We encode the source address to bytes
const sourceAddressBytes = getAddressEncoder().encode(signedByAddress);
// Then we create a public Ed25519 key with those bytes
const signedByPublicKey = await crypto.subtle.importKey('raw', sourceAddressBytes, 'Ed25519', true, ['verify']);
// Now we can verify the signature using that key
const verifiedSignature = await verifySignature(signedByPublicKey, signedBySignature, decodedTransaction.messageBytes);
```

We could write a helper function that takes as input the bytes purported to have been signed, signature to verify and address, and performs the verification for you.

## Example use case

```ts
const signedByAddress = 'ED1WqT2hWJLSZtj4TtTdoovmpMrr7zpkUdbfxmcJR1Fq' as Address;
const signedBySignature = decodedTransaction.signatures[signedByAddress]!;
const verifiedSignature = await verifySignatureForAddress(signedByAddress, signedBySignature, decodedTransaction.messageBytes)
```

## Details

- This should just be the code from my example - address to bytes, verify-only public key from bytes, `verifySignature` call
- We may want to note in documentation that you shouldn't do this if you're going to re-use the key for multiple verifications since it'll re-create the same key multiple times 

Alternatively, perhaps we could create a function eg `createVerifyPublicKeyForAddress` that just gives you the public key for an address, to avoid that potential performance issue?

**Comment by steveluscher** [comment]
I agree that we shouldn't create a method that internally constructs the public key only to throw it out, lest that be called more than once for the same address. Creating an address-to-public key helper sounds fine though. No need to use ‘verify’ in that function name, since all Ed25519 `CryptoKeys` have only `['verify']` usage.

**Comment by andiz2** [comment]
I can take care of this
